src/main/java/uk/gov/hmcts/reform/sscs/service/RoboticsJsonUploadService.java

```diff
@@ -0,0 +1,129 @@
+
+        log.info("Attaching Robotics JSON to case {}", caseDetails.getId());
+        SscsCaseData updatedCaseData = attachRoboticsJsonToCaseData(caseData, uploadResponse);
+
+        try {
+            ccdService.updateCase(updatedCaseData, caseDetails.getId(),
+                    "attachRoboticsJson", "", "", idamTokens);
+        } catch (CcdException ccdEx) {
+            log.error("Failed to update ccd case with Robotics JSON but carrying on [" + caseDetails.getId() + "] ["
+                    + caseData.getCaseReference() + "]", ccdEx);
+        }
+
+    }
+
+    public SscsCaseData attachRoboticsJsonToCaseData(SscsCaseData caseData, UploadResponse uploadResponse) {
+        DocumentLink documentLink = getDocumentLink(uploadResponse);
+
+        log.info("Robotics JSON Document for CCD Case {} uploaded to {}", caseData.getCcdCaseId(), documentLink);
+
+        SscsDocument roboticsJsonDocument = getRoboticsJsonDocument(documentLink);
+
+        log.info("Adding Robotics JSON document to CCD Case Id {}", caseData.getCcdCaseId());
+
+        List<SscsDocument> sscsDocumentList = updateCaseDataDocuments(caseData, roboticsJsonDocument);
+        caseData.setSscsDocument(sscsDocumentList);
+
+        return caseData;
+    }
+
+    private SscsDocument getRoboticsJsonDocument(DocumentLink documentLink) {
+        SscsDocumentDetails roboticsJsonDocumentDetails = SscsDocumentDetails.builder()
+                .documentFileName(ROBOTICS_JSON_FILENAME)
+                .documentDateAdded(LocalDateTime.now().format(DateTimeFormatter.ISO_DATE))
+                .documentLink(documentLink)
+                .build();
+
+        return new SscsDocument(roboticsJsonDocumentDetails);
+    }
+
+    private DocumentLink getDocumentLink(UploadResponse uploadResponse) {
+        final String href = uploadResponse.getEmbedded().getDocuments().get(0).links.binary.href;
+        return DocumentLink.builder().documentUrl(href).build();
+    }
+
+    private List<SscsDocument> updateCaseDataDocuments(SscsCaseData caseData, SscsDocument roboticsJsonDocument) {
+        List<SscsDocument> sscsDocumentList = caseData.getSscsDocument();
+        if (CollectionUtils.isEmpty(sscsDocumentList)) {
+            sscsDocumentList = singletonList(roboticsJsonDocument);
+        } else {
+            sscsDocumentList.add(roboticsJsonDocument);
+        }
+        return sscsDocumentList;
+    }
+
+    public UploadResponse uploadRoboticsJson(List<MultipartFile> files) {
+
+        String serviceAuthorization = authTokenGenerator.generate();
+
+        try {
+            return documentUploadClientApi
+                    .upload(S2S_TOKEN, serviceAuthorization, DM_STORE_USER_ID, files);
+        } catch (HttpClientErrorException httpClientErrorException) {
+            log.error("Doc Store service failed to upload Robotics JSON", httpClientErrorException);
+            throw new UnsupportedDocumentTypeException(httpClientErrorException);
```

Comment:
Please, notice that if the system fails to upload the json then it throws an exception that bubbles up to the end users. I don't think we want the user to know that we fail to upload the robotic JSON to the DocStore, do we? 
I think we just need to catch any Exception and log it as an info message. 
Make sure, this method is not used somewhere else as the error handling then required there might be different

Reply:
Yes, that's a good point. We don't want the system to report technical difficulties in this situation.